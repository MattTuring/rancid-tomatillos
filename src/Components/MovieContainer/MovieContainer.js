import React from 'react'
import {connect} from 'react-redux'
import MovieCard from '../MovieCard/MovieCard'

const MovieContainer = ({movies}) => {

console.log(movies)


  return(
    <div className='movie-container'>
    {movies[0] && movies.map(film => {
      console.log(film)
      return (
      <MovieCard
      key={film.id}
      title={film.title}
      poster={film.poster_path}
      rating={film.rating}
      />
    )
  })}
    </div>
  )
}

const mapStateToProps = state => ({
  movies: state.movies
})

export default connect(mapStateToProps)(MovieContainer)
