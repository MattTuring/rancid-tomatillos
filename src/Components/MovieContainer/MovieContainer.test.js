import React from 'react';
import { shallow } from 'enzyme';
import { MovieContainer } from './MovieContainer';

describe('MovieContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<MovieContainer
      movies={[{  id:7, title: 7,
      poster_path:7,
        average_rating:7}]}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
