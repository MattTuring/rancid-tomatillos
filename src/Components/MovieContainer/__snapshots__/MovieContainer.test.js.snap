// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should match the snapshot 1`] = `
<div
  className="movie-container"
>
  <Connect(MovieCard)
    averageRating={7}
    id={7}
    key="7"
    poster={7}
    title={7}
  />
</div>
`;
